### DISCUSSION TITLE
SetOption114, dedicated MQTT topic and toggle state

### DISCUSSION THREAD

**Discussion by merlinschumacher** [arendst/Tasmota#17711] [Q&A] [answered]
Hi,

I have multiple Shelly 1s that should send a `TOGGLE` MQTT message whenever the connected switch is changed, and only then. Using SetOption114 and SwitchMode15 Tasmota always sends `ON` and `OFF`. Additionally, the message is sent on `tele/DEVICE/sensor` and repeated every 5 minutes, but not on a specific topic like `cmnd` or `stat`. I need an MQTT message only when the button is hit and not again every few minutes. Additionally having a `TOGGLE` instead of the specific state would be nice.
Does anyone have an idea how to achieve this?

Thanks for your help!

**Comment by barbudor** [comment, marked as answer]
Have you looked at the switch topic command?

Alternatively you can use a rule to republish the on or off event to another topic

**Reply by merlinschumacher** [reply]
`SwitchTopic` doesn't seem to change anything. 
I've tried the following rule and it didn't work: 
`Rule on Switch1#state=1 do publish stat/test/SWITCH1 ON endon on Switch1#state=0 do publish stat/test/SWITCH1 OFF endon`

**Reply by sfromis** [reply]
If you want to use rules, do not set `SwitchMode 15` which says to **only** publish, meaning no default actions, and not available to rules. More generally, I'd recommend to not just add more than one option at the same time, unless you specifically know and want what the combination does.

**Reply by merlinschumacher** [reply]
Thanks for hint with `SwitchMode 15` disabling rules. Setting it back to `1` enabled the rules, and I made a rule that just sends a `TOGGLE` when hitting my rocker switch.
